migrator/init.lua

```diff
@@ -60,21 +62,54 @@ local function get_storage_timeout()
     return DEFAULT_STORAGE_TIMEOUT
 end
 
+-- Makes sure that the passed migrations match the list from the local reader.
+local function check_migrations_consistency(migrations_per_instance)
+    local names = fun.iter(vars.loader:list()):map(function(m) return m.name end):totable()
+    for host, applied in pairs(migrations_per_instance) do
+        if utils.compare(names, applied) == false then
+            local err_msg = string.format('Inconsistent migrations in cluster: '
+            .. 'expected: %s, applied on %s: %s', json.encode(names), host, json.encode(applied))
+            log.error(err_msg)
+            error(err_msg)
+        end
+    end
+end
+
+-- Makes sure there is no migrations list in cluster config.
+local function check_no_migrations_in_config()
+    local config = confapplier.get_readonly('migrations')
+    if config ~= nil and config.applied ~= nil and #config.applied > 0 then
+        error('Cannot perform an upgrade. A list of applied migrations is found in cluster ' ..
+            'config. Current migrator version works only with local list of applied migrations. ' ..
+            'Run "move_migrations_state" to move cluster-wide migrations state to local ' ..
+            'storage before up invocation.')
+    end
+end
+
+-- Returns server alias by URI.
+local function get_server_alias(instance_uri)
+    local servers = get_topology_api.get_servers(function(server)
+        return server.uri == instance_uri
+    end)
+    if not servers or #servers == 0 or #servers > 1 or not servers[1].alias then
```

Comment:
I would not like to raise an error if we are not able to get an alias. We do have a URI, which we can use instead of an alias. This is not public API, so I think this is acceptable for this function.

Reply:
Well, ok, let it be